lib/LaTeXML/Package/physics.sty.ltxml

```diff
@@ -467,11 +555,12 @@ DefMacro('\matrixelement', sub {
     my $arg0 = $gullet->readArg();
     my $arg1 = $gullet->readArg();
     my $arg2 = $gullet->readArg();
-    my ($cargs, $pargs) = dualize_arglist('#1#2#3', $arg1, $arg0, $arg2);    # NOTE order! op first
-    return Invocation(T_CS('\DUAL'), undef,
-      Invocation(T_CS('\@APPLY'), Tokens($cfunc, grep { $_ } @$cargs)),
-      Invocation(T_CS('\@WRAP'),
-        Tokens($open, $$pargs[1], $middle, $$pargs[0], $middle, $$pargs[2], $close))); });
+    return BuildXMDual(
+      { reversion => Tokens(T_CS('\matrixelement'), ($size ? () : T_OTHER('*')),
+          phys_revArg(T_XMArg(1)), phys_revArg(T_XMArg(2)), phys_revArg(T_XMArg(3))) },
+      Invocation(T_CS('\@APPLY'), Tokens($cfunc, T_XMArg(2), T_XMArg(1), T_XMArg(3))),
+      Tokens($open, T_XMArg('1'), $middle, T_XMArg(2), $middle, T_XMArg(3), $close),
+      $arg0, $arg1, $arg2); });
```

Comment:
Somehow the new diff is less readable than the old Invocation-based return value. Maybe it's a limitation of perltidy indenting the invocation calls, just marking that in passing.

Reply:
Maybe; important thing is to learn the api (keywords, content, presentation).